packages/tc-ui/src/primitives/relationship/lib/relationship.jsx

```diff
@@ -1,25 +1,96 @@
+/* global window */
 const React = require('react');
+const { RichRelationships } = require('./rich-relationship');
 
-const Relationship = ({ value, disabled, onRelationshipRemove, index }) => (
-	<li
-		data-name={value.name}
-		data-code={value.code}
-		className="selected-relationship"
-		key={index}
-	>
-		<button
-			type="button"
-			disabled={disabled ? 'disabled' : null}
-			className={`o-buttons o-buttons--small relationship-remove-button ${
-				disabled ? 'disabled' : ''
-			}`}
-			onClick={onRelationshipRemove}
-			data-index={index}
-		>
-			Remove
-		</button>
-		<span className="o-layout-typography">{value.name || value.code}</span>
-	</li>
-);
+class Relationship extends React.Component {
+	constructor(props) {
+		super();
+		this.props = props;
+		this.state = {
+			annotate: false,
+			isMounted: false,
+		};
 
+		this.showRichRelationshipEditor = this.showRichRelationshipEditor.bind(
+			this,
+		);
+	}
+
+	componentDidMount() {
+		this.setState(prevState => {
+			const { isMounted } = prevState;
+			const oldState = isMounted;
+			return {
+				isMounted: !oldState,
+			};
+		});
+	}
+
+	showRichRelationshipEditor(event) {
+		this.setState(
+			prevState => {
+				const { annotate } = prevState;
+				const oldState = annotate;
+				return {
+					annotate: !oldState,
+				};
+			},
+			() => {
+				// TODO - can this be done without using global window
+				if (window && this.state.isMounted) {
+					window.Origami['o-expander'].init();
+				}
+			},
+		);
+		event.stopPropagation();
+	}
+
+	render() {
+		const { value, disabled, onRelationshipRemove, index } = this.props;
+
+		const { isMounted, annotate } = this.state;
+
+		return (
+			<>
+				<li
+					data-name={value.name}
+					data-code={value.code}
+					className="selected-relationship"
+					key={index}
+				>
+					<span>
+						<button
+							type="button"
+							disabled={disabled ? 'disabled' : null}
+							className={`o-buttons o-buttons--small relationship-remove-button ${
+								disabled ? 'disabled' : ''
+							}`}
+							onClick={onRelationshipRemove}
+							data-index={`remove-${index}`}
+						>
+							Remove
+						</button>
+						<span className="o-layout-typography">
+							{value.name || value.code}
+						</span>
+						<button
+							type="button"
+							disabled={disabled ? 'disabled' : null}
+							className={`o-buttons o-buttons--small relationship-annotate-button ${
+								disabled ? 'disabled' : ''
+							}`}
+							onClick={this.showRichRelationshipEditor}
+							data-index={`annotate-${index}`}
+						>
+							Annotate
+						</button>
+					</span>
+					{isMounted && annotate ? (
+						<RichRelationships key={index} {...this.props} />
```

Comment:
Small niggle - can this component be renamed to RichRelationship or RelationshipProperties or something. While the implementation was in flux I wasn't sure what the content of the component would end up being, but now (if I understand correctly) it is a wrapper around the properties of a single relationship

Reply:
`ok, I think I follow. Do we have cypress tests for his scenario?`
Yes =>
https://github.com/Financial-Times/treecreeper/blob/6352deb1ef8352ca81cc55b447554fcf27cd3dc5/packages/tc-ui/src/pages/edit/__tests__/e2e-edit-record.cyp.js#L110

`Have you manually tested that the bad behaviour described in the comment doesn't happen after this change?`
I did and didn't see that behaviour happening ... it might be, in addition, because we have fixed the issue we had with Remove button responding to other clicks inside of `<label>` region